overlord/state/taskrunner.go

```diff
@@ -94,6 +102,7 @@ func (r *TaskRunner) run(t *Task) {
 		panic("internal error: attempted to run task with nil handler for status " + t.Status().String())
 	}
 
+	t.scheduledTime = time.Time{} // clear schedule
```

Comment:
Indeed, you are right. I was concerned it'd not get scheduled correctly on a re-run.